blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeTest.java

```diff
@@ -2825,30 +3037,60 @@ public void orphanParallels2() throws Exception {
             if (i == 1) {
                 assertEquals("Parallel", n.get("displayName"));
                 assertEquals(nodes.get(i + 1).get("id") + "-parallel-synthetic", n.get("id"));
-                Assert.assertEquals(3, edges.size());
+                assertEquals(3, edges.size());
                 assertEquals(nodes.get(i + 1).get("id"), edges.get(0).get("id"));
                 assertEquals(nodes.get(i + 2).get("id"), edges.get(1).get("id"));
                 assertEquals(nodes.get(i + 3).get("id"), edges.get(2).get("id"));
             }
             if (i == 2) {
                 assertEquals("branch1", n.get("displayName"));
                 assertEquals(1, edges.size());
-                assertEquals(nodes.get(5).get("id"), edges.get(0).get("id"));
+                assertEquals(nodes.get(6).get("id"), edges.get(0).get("id"));
             }
             if (i == 3) {
                 assertEquals("branch2", n.get("displayName"));
                 assertEquals(1, edges.size());
-                assertEquals(nodes.get(5).get("id"), edges.get(0).get("id"));
+                assertEquals(nodes.get(10).get("id"), edges.get(0).get("id"));
             }
             if (i == 4) {
                 assertEquals("branch3", n.get("displayName"));
                 assertEquals(1, edges.size());
-                assertEquals(nodes.get(5).get("id"), edges.get(0).get("id"));
+                assertEquals(nodes.get(8).get("id"), edges.get(0).get("id"));
             }
             if (i == 5) {
                 assertEquals("stage2", n.get("displayName"));
                 assertEquals(0, edges.size());
             }
+            if (i == 6) {
+                assertEquals("Setup", n.get("displayName"));
+                assertEquals(1, edges.size());
+                assertEquals(nodes.get(7).get("id"), edges.get(0).get("id"));
+            }
+            if (i == 7) {
+                assertEquals("Unit and Integration Tests", n.get("displayName"));
+                assertEquals(1, edges.size());
+                assertEquals(nodes.get(5).get("id"), edges.get(0).get("id"));
+            }
+            if (i == 8) {
+                assertEquals("Setup", n.get("displayName"));
+                assertEquals(1, edges.size());
+                assertEquals(nodes.get(9).get("id"), edges.get(0).get("id"));
+            }
+            if (i == 9) {
+                assertEquals("Unit and Integration Tests", n.get("displayName"));
+                assertEquals(1, edges.size());
+                assertEquals(nodes.get(5).get("id"), edges.get(0).get("id"));
```

Comment:
By the way, this sort of test code is screaming for an assertion utility, something like

```java
assertEdges("Pipeline1#1", …, "Setup #9", "Unit and Integration Tests #5", …);

Reply:
totally, I was just following the pattern already there, since I wasn't sure if you guys are open to refactoring this kind of stuff...

Reply:
@countableSet definitely happy to merge this if you want to do it